app/core/middleware/logger.py

```diff
@@ -1,59 +1,125 @@
 import json
+import logging
 import time
+from typing import Callable
+from uuid import uuid4
+from fastapi import FastAPI, Response
 from starlette.types import Message
 from starlette.requests import Request
 from starlette.middleware.base import BaseHTTPMiddleware
-from app.core.middleware.iterator_wrapper import iterator_wrapper as aiwrap
+from app.core.utils.async_iterator_wrapper import AsyncIteratorWrapper
 
 class LogMiddleware(BaseHTTPMiddleware):
 
-    # Leaving the code in here that sets up the request and response body just in
-    # case we want to use it in the future. But for now it's taking up way too much
-    # log space for no benefit.
+    def __init__(self, app: FastAPI, *, logger: logging.Logger, file_logger: logging.Logger = None) -> None:
+        self._logger = logger
+        self._file_logger = file_logger
+        super().__init__(app)
 
     # https://github.com/tiangolo/fastapi/issues/394#issuecomment-927272627
-    # async def set_body(self, request: Request):
-    #     receive_ = await request._receive()
+    async def set_body(self, request: Request):
+        receive_ = await request._receive()
 
-    #     async def receive() -> Message:
-    #         return receive_
+        async def receive() -> Message:
+            return receive_
 
-    #     request._receive = receive
+        request._receive = receive
 
     async def dispatch(self, request, call_next):
-            start_time = time.time()
+        if request.url.path.startswith("/api/v1/health"):
+            response = await self._execute_request(call_next, request)
+            return
+        
+        request_id: str = str(uuid4())
+        logging_dict = {
+            "X-API-REQUEST-ID": request_id  # X-API-REQUEST-ID maps each request-response to a unique ID
+        }
 
-            # await self.set_body(request)
-            # req_body = await request.body()
-            response = await call_next(request)
+        await self.set_body(request)
+        response, response_dict = await self._create_response_log(
+            call_next,
+            request,
+            request_id
+        )
+        request_dict = await self._create_request_log(request)
+        logging_dict["req"] = request_dict
+        logging_dict["res"] = response_dict
```

Comment:
you could just be setting this in the literal construct it seems like?

Reply:
not sure what you mean by this

Reply:
You could set these fields where you declare the dictionary.